### PR TITLE
add APIs to execution context for middleware support

### PR DESCRIPTION
**Pull request by kaibocai**
add APIs to execution context for middleware support

### CODE REVIEW

**Comment by patriot1burke** [comment]
@kaibocai If you add support for the ExecutionContext methods to the Java worker, I can test them out in the Quarkus extension I wrote to integration with azure functions.

File: src/main/java/com/microsoft/azure/functions/ExecutionContext.java [hunk 1]
```diff
@@ -50,5 +52,19 @@ default TraceContext getTraceContext() {
...
+
+    Map<String, String> getParameterPayloadMap();
+
+    void setMiddlewareInput(String key, Object value);
+
+    Object getReturnValue();
+
+    void setMiddlewareOutput(Object value);
+
+    void setFunctionInstance(Object functionInstance);
```
Comment (patriot1burke) [inline src/main/java/com/microsoft/azure/functions/ExecutionContext.java:66]: Might consider moving these middleware specific methods like setFunctionInstance to a separate interface that extends extends ExecutionContext.  ExecutionContext is available to the application developer and these methods might not be something you want exposed to them.  See my PR here which shows this:

https://github.com/Azure/azure-functions-java-core-library/pull/10

Comment (kaibocai) [reply src/main/java/com/microsoft/azure/functions/ExecutionContext.java:66]: Thank you for your valuable suggestion! We will discuss it with the team.

**Comment by kaibocai** [comment]
> @kaibocai If you add support for the ExecutionContext methods to the Java worker, I can test them out in the Quarkus extension I wrote to integration with azure functions.

Hi @patriot1burke , please use [this java worker](https://github.com/Azure/azure-functions-java-worker/tree/kaibocai/f-dc-middleware) and this pr core library for local testing. 

`mvn clean install -Dgpg.skip` can install this core library to local maven repo. Thank you.

**Comment by kaibocai** [comment]
Refer to https://github.com/Azure/azure-functions-java-core-library/pull/10#discussion_r969946755. 
Need to check if function instance has already been set.

**Pull request closed without merging**